src/main/java/com/zerobase/mentalgrowhdiary/repository/impl/FeedbackRepositoryImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.zerobase.mentalgrowhdiary.repository.impl;
+
+import com.querydsl.core.types.Projections;
+import com.querydsl.jpa.JPAExpressions;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import com.zerobase.mentalgrowhdiary.domain.*;
+import com.zerobase.mentalgrowhdiary.dto.FeedbackWithLatestReservationDTO;
+import com.zerobase.mentalgrowhdiary.repository.custom.FeedbackRepositoryCustom;
+import com.zerobase.mentalgrowhdiary.type.FeedbackStatus;
+import com.zerobase.mentalgrowhdiary.type.ReservationStatus;
+import lombok.RequiredArgsConstructor;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@RequiredArgsConstructor
+public class FeedbackRepositoryImpl implements FeedbackRepositoryCustom {
+
+    private final JPAQueryFactory jpaQueryFactory;
+
+    @Override
+    public List<FeedbackWithLatestReservationDTO> getFeedbackWithLatestReservation(String username, String feedbackStatus, String reservationStatus,
+                                                                                   LocalDateTime reservationDate) {
+
+        QFeedback qFeedback = QFeedback.feedback;
+        QDiary qDiary = QDiary.diary;
+        QReservation qReservation = QReservation.reservation;
+        QUser qUser = QUser.user;
+
+        QReservation r2 = new QReservation("r2");
+
+
+        return jpaQueryFactory
+                .select(Projections.constructor(
+                        FeedbackWithLatestReservationDTO.class,
+                        qFeedback.feedbackId,
+                        qFeedback.feedbackStatus.stringValue(),
+                        qFeedback.createdDate,
+                        qDiary.diaryId,
+                        qDiary.updatedDate,
+                        qReservation.reservationId,
+                        qReservation.reservationDateTime,
+                        qReservation.status.stringValue()
+                ))
+                .from(qFeedback)
+                .join(qDiary).on(qFeedback.diary.eq(qDiary))
+                .join(qUser).on(qDiary.user.eq(qUser))
+                .leftJoin(qReservation).on(qReservation.user.eq(qUser)
+                        .and(qReservation.reservationDateTime.eq(
+                                JPAExpressions.select(r2.reservationDateTime.max())
+                                        .from(r2)
+                                        .where(r2.user.eq(qUser))
+                        )))
+                .where(
+                        username != null ? qUser.username.containsIgnoreCase(username) : null,
```

Comment:
삼항 연산자는 가독성을 떨어뜨리므로 이 부분을 method로 빼서 넣어도 좋을 것 같아요. 
```suggestion
.where(containUserNameIgnoreCase(username), 

...

 private BooleanExpression containUserNameIgnoreCase(String usename) {
        if(usename == null) {
            return null;
        }
        return qUser.username.containsIgnoreCase(username);
    }
```

Reply:
감사합니다. 해당 로직으로 수정하였습니다.